File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/android/src/main/kotlin/com/danhdue/{{name.snakeCase()}}/domain/repository/{{name.pascalCase()}}Repository.kt
/*
 * Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.
 */

package com.danhdue.{{name.snakeCase()}}.domain.repository

import com.danhdue.{{name.snakeCase()}}.domain.model.{{name.pascalCase()}}Data

interface {{name.pascalCase()}}Repository {
    suspend fun getData(): Result<{{name.pascalCase()}}Data>
    suspend fun syncData(): Result<Boolean>
}

File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/ios/{{name.snakeCase()}}/Sources/{{name.snakeCase()}}/Domain/Repository/{{name.pascalCase()}}Repository.swift
// Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.

import Foundation

/// Repository protocol for plugin domain operations.
public protocol {{name.pascalCase()}}Repository: Sendable {
    func getData() async throws -> {{name.pascalCase()}}Data
    func syncData() async throws
}
